keras/models.py

```diff
@@ -741,28 +741,59 @@ def save_weights(self, filepath, overwrite=True):
 
     def compile(self, optimizer, loss,
                 metrics=None,
+                loss_weights=None,
                 sample_weight_mode=None,
                 weighted_metrics=None,
+                target_tensors=None,
```

Comment:
We should consider naming it `target_tensor` since it will always be a single tensor.

Reply:
Same as above. Maybe we should warn/assert that len(target_tensors)==1, to make a sequential Model and Sequential interchangeable

Reply:
When passing multiple target_tensors to this function, a very clear exception is raised:
```
Traceback (most recent call last):
  File "./examples/cifar10_cnn.py", line 68, in <module>
    target_tensors=[1,2])
  File "build/bdist.linux-x86_64/egg/keras/models.py", line 817, in compile
  File "build/bdist.linux-x86_64/egg/keras/engine/training.py", line 715, in compile
ValueError: When passing a list as `target_tensors`, it should have one entry per model
outputs. The model has 1 outputs, but you passed target_tensors=[1, 2]
```

Reply:
It' mainly to make a sequential Model and Sequential interchangeable as much as practicable. For a sequential Model compile() we (correctly) don't prohibit loss_weights, and (correctly) have target_tensors. When passing multiple loss_weights or target_tensors, a very clear exception is raised:
```The model has 1 outputs, but you passed loss_weights=[1, 2]``` and 
```The model has 1 outputs, but you passed target_tensors=[1, 2]```

Reply:
Ok for this change.